esx_service/vmdk_ops.py

```diff
@@ -402,8 +402,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
 # Return error, or None for OK
 def removeVMDK(vmdk_path):
     logging.info("*** removeVMDK: %s", vmdk_path)
-    cmd = "{0} {1}".format(VMDK_DELETE_CMD, vmdk_path)
 
+    # Check the current volume status
+    attached, uuid, attach_as = getStatusAttached(vmdk_path)
+    if attached:
```

Comment:
@brunotm, please call handle_stale_attach() if the volume is marked as attached. That is a better check than just to see if the volume is attached. If attached and the attaching VM is powered-off then go ahead clean up the volume config and then remove the VMDK.